File: testsuite/tests/ada/0009-exceptions/cpp/main.cpp
#include <iostream>

#include "test.h"
#include "other.h"

int main() {

    try {
        test::get_exception(1);
    } catch (const test::Exc1 &e) {
        std::cout << "caught" << e.what() << "\n";
    }

    try {
        test::get_exception(2);
    } catch (const test::Exc2 &e) {
        std::cout << "caught" << e.what() << "\n";
    }

    try {
        test::get_exception(3);
    } catch (const polyglot::exceptions::polyglot_exception &e) {
        std::cout << "caught" << e.what() << "\n";
    }

    try {
        test::get_exception(0);
    } catch (const other::Exc1 &e) {
        std::cout << "caught" << e.what() << "\n";
    }

    // should not throw: verify that the exception was cleared.
    test::get_exception(4);
}

File: testsuite/tests/ada/0009-exceptions/java/Main.java
import com.adacore.libtest.test.TestPackage;
import com.adacore.libtest.test.Exc1;
import com.adacore.libtest.test.Exc2;
import com.adacore.gnatpolyglot.runtime.PolyglotException;

public class Main {

    public static void main(String[] args) throws Throwable {
        try {
            TestPackage.getException(1);
        } catch (Exc1 e) {
            System.out.println("caught" + e.getMessage());
        }
        try {
            TestPackage.getException(2);
        } catch (Exc2 e) {
            System.out.println("caught" + e.getMessage());
        }
        try {
            TestPackage.getException(3);
        } catch (PolyglotException e) {
            System.out.println("caught" + e.getMessage());
        }
        try {
            TestPackage.getException(0);
        } catch (com.adacore.libtest.other.Exc1 e) {
            System.out.println("caught" + e.getMessage());
        }

        // should not throw: verify that the exception was cleared.
        TestPackage.getException(4);
    }
}
